pyblish_qml/control.py

```diff
@@ -353,29 +353,24 @@ def iterator(self, plugins, context):
                     signals.pop(order).emit()
 
             if not self.data["state"]["is_running"]:
-                return StopIteration("Stopped")
+                return
 
             if test(**state):
                 self.data["state"]["testPassed"] = False
-                return StopIteration("Stopped due to %s" % test(**state))
+                raise RuntimeError("Stopped due to %s" % test(**state))
 
             self.data["state"]["testPassed"] = True
 
-            try:
-                # Notify GUI before commencing remote processing
-                self.about_to_process.emit(plug, instance)
-
-                result = self.host.process(plug, context, instance)
+            # Notify GUI before commencing remote processing
+            self.about_to_process.emit(plug, instance)
 
-            except Exception as e:
-                return StopIteration("Unknown error: %s" % e)
+            result = self.host.process(plug, context, instance)
```

Comment:
Is it safe to remove this try/except block? self.host.process can throw an error, in which case we aren't handling it properly. That's what the try/except was doing. Also, why are we removing it? Better do another `yield` and `return` here.

Reply:
It is safe as all exception would be caught in the `util.defer` logic and transmitted to the `on_next` callback as an Exception object. However as you mentioned in your other [comment](https://github.com/pyblish/pyblish-qml/pull/362/files#r389347317), it means that all exceptions would be routed in the `on_finished` callback which is not great.

Reply:
Reverted